src/rules.jl

```diff
@@ -170,29 +172,34 @@ end
     :NaN, :(  (SpecialFunctions.bessely($ν - 1, $x) - SpecialFunctions.bessely($ν + 1, $x)) / 2  )
 @define_diffrule SpecialFunctions.besselk(ν, x)   =
     :NaN, :( -(SpecialFunctions.besselk($ν - 1, $x) + SpecialFunctions.besselk($ν + 1, $x)) / 2  )
+@define_diffrule SpecialFunctions.besselkx(ν, x)  =
+    :NaN, :( -(SpecialFunctions.besselkx($ν - 1, $x) - 2SpecialFunctions.besselkx($ν, $x) + SpecialFunctions.besselkx($ν + 1, $x)) / 2 )
 @define_diffrule SpecialFunctions.hankelh1(ν, x)  =
     :NaN, :(  (SpecialFunctions.hankelh1($ν - 1, $x) - SpecialFunctions.hankelh1($ν + 1, $x)) / 2  )
+@define_diffrule SpecialFunctions.hankelh1x(ν, x) =
+    :NaN, :(  (SpecialFunctions.hankelh1x($ν - 1, $x) - 2im*SpecialFunctions.hankelh1x($ν, $x) - SpecialFunctions.hankelh1x($ν + 1, $x)) / 2  )
 @define_diffrule SpecialFunctions.hankelh2(ν, x)  =
     :NaN, :(  (SpecialFunctions.hankelh2($ν - 1, $x) - SpecialFunctions.hankelh2($ν + 1, $x)) / 2  )
+@define_diffrule SpecialFunctions.hankelh2x(ν, x)  =
+    :NaN, :(  (SpecialFunctions.hankelh2x($ν - 1, $x) + 2im*SpecialFunctions.hankelh2x($ν, $x) - SpecialFunctions.hankelh2x($ν + 1, $x)) / 2  )
```

Comment:
And here (https://github.com/JuliaMath/SpecialFunctions.jl/blob/1febdd05902f5b3a3adc9915aba40b776aa3ea24/src/chainrules.jl#L137):
```suggestion
    :NaN, :(  (SpecialFunctions.hankelh2x($ν - 1, $x) - SpecialFunctions.hankelh2x($ν + 1, $x)) / 2 + im * SpecialFunctions.hankelh2x($ν, $x)  )
```